### ISSUE TITLE
Apple’s deep learning frameworks: BNNS vs. Metal CNN

### ISSUE THREAD

**Issue by mmoaay** [SwiftGGTeam/translation#44]
[http://machinethink.net/blog/apple-deep-learning-bnns-versus-metal-cnn/](http://machinethink.net/blog/apple-deep-learning-bnns-versus-metal-cnn/)

**Comment by Desgard** [comment]
原文： **performing inference** 在一般教科书中译为 **变分推断**。

推荐修改：

> 目前 BNNS 和 MPSCNN 在**卷积神经网络**领域中用于**执行推断**。

改为

> 目前 BNNS 和 MPSCNN 在**卷积神经网络**领域中用于**变分推断**。

**Comment by Desgard** [comment]
原文：

> Once you have a trained network, you can use it to make predictions. This is called “inference”. While training requires a number of dedicated, heavy-duty computers, doing inference is perfectly possible on a modern mobile phone.

--- 

原翻译：

> 一旦网络训练完成，便可以用来进行**预测**。这被称为“推理”。虽然培训需要一些专用的重型计算机，但在现代的手机上进行推理是完全可能的。

建议修改为：

> 当得到一个训练网络，便可以用来进行**预测**。这被称为“推断”。训练本应需要使用重型计算机，但在现代的手机上进行推断是完全可能的。

--- 

将 **predictions** 译成推断，更符合统计学教科书中的译法。例如：贝叶斯推断（Bayesian inference）等

**Comment by Desgard** [comment]
原翻译：

> 已经提供的建构单元（卷积层，池化层和完全连接层）

建议修改为：

> 已经提供的建构单元（卷积层，池化层和全连接层）

**Connected Layer** 一般译为全连接层。

**Comment by Desgard** [comment]
原文：

> * Normalization (spatial, cross channel, local contrast)
> * Softmax
> * Logarithmic softmax
> * You can use the activation functions as layers

原翻译：

> * 规范化层（）
> * Softmax
> * 对数 Softmax
> * 使用激活函数作为层级

建议修改为：

> * 归一化（特征归一化、跨通道归一化（弱化）、局部对比度归一化）
> * Softmax，也称为归一化指数函数
> * 对数 Softmax，即使用 Softmax 函数并配合 log 似然代价函数
> * 激活函数层

**Comment by Desgard** [comment]
`Pokémon` 建议不翻译！！！😝

**Comment by Desgard** [comment]
原文：
> The app creates a neural network with 2 convolutional layers, 1 max pooling layer, 1 averaging pooling layer, and a fully-connected layer. 

原翻译：
> 该 App 创建了具有 2 个卷积层，1 个最大池层，1 个平均池层和 1 个完全连接层的神经网络。

建议修改为：
> 这个 App 创建的神经网络具有 2 个卷积层、1 个 max-pooling 层，1 个 average-pooling 层和 1 个全连接层。

**Comment by liberalisman** [comment]
原文：
> So… as iOS developers we now have two APIs for deep learning that appear to do pretty much the same thing.
原翻译：
>  所以，作为 iOS 开发者，现在有了两个用来做深度学习的 APIs，并且它们看起来很相似。
建议翻译:
> 所以，作为 iOS 开发者，有了两个用于做深度学习的框架，它们有很多类似的地方。
